### PR TITLE
span_links: refactor tests

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

Makes tests easier to understand and maintain

## Changes

- Ensures Span Links are not initialized with a parent_id and http_headers. Only one should be used.
- Avoids generating span links between nested spans (parent-child). This is relationship does not reflect realworld applications. 
- Renames spans and variables to more meaningful values.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Ci is failing, I've converted the PR to draft.

You can set it back to normal once you need a review

**Comment by robertomonteromiguel** [comment]
Moved to draft. There are parametric failures, related with this PR ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
CI failures need to be fixed :)

**Review by cbeauchesne — APPROVED** [review]
From framework usage, it's ok.

It'd be a good idea to request a review from somebody  familiar with the tested feature.

File: manifests/nodejs.yml [hunk 1]
```diff
@@ -401,7 +401,6 @@ tests/:
       TestDynamicConfigV1: v4.11.0
       TestDynamicConfigV1_ServiceTargets: missing_feature
       TestDynamicConfigV2: v4.23.0
```
Comment (mabdinur) [inline manifests/nodejs.yml:403]:
```suggestion
      TestDynamicConfigV2: v4.23.0
    test_span_links.py: missing_feature
```

**Pull request merged**